include/aspect/utilities.h

```diff
@@ -1011,6 +1012,140 @@ namespace aspect
     };
 
 
+
+    /**
+     * UnstructuredDataLookup reads in files containing input data in ascii format.
+     * Note the required format of the input data: The first lines may contain
+     * any number of comments if they begin with '#', but one of these lines
+     * needs to contain the number of columns and lines in the file:
+     * '#POINTS: TOTALNUMBERDATALINES NUMBERCOORDCOLUMNS NUMBERDATACOLUMNS'.
+     * For example '# POINTS: 5 4 3' would be 5 lines of data, 4 columns of coordinates,
+     * and 3 columns of data. The comments can optionally be followed by a
+     * single line, which does not start with '#', containing the names of
+     * the data columns.
+     * The order of the following data columns has to be
+     * 'coordinates data' with @p querydim coordinate columns and @p components
+     * data columns. @p querydim and @p components need to
+     * match NUMBERCOORDCOLUMNS NUMBERDATACOLUMNS, respectivily.
+     * The coordinates do not need to be equidistant.
+     */
+    template <int querydim>
+    class UnstructuredDataLookup
+    {
+      public:
+
+        /**
+         * The following are member variables and functions of the class
+         */
+        unsigned int n_data_files;
+
+        std::vector< Point<querydim> > coordinate_values;
+
+        std::vector< std::vector<double> > data;
+
+
+        // Create an instance of a kdtree object as a member variable of the class
+        KDTree<querydim> coordinate_kdtree;
+
+
+        virtual void initialize (const MPI_Comm &comm,
+                                 std::string datadirectory,
+                                 std::vector<std::string> file_name_list);
+
+
+
+        /**
+         * Constructor that explicitly prescribes the number of data columns
+         * in the data file. If a list of data components is provided in the
+         * data file it is checked that the length of this list is consistent
+         * with this number of components. This constructor is mostly provided
+         * for backwards compatibility. Not prescribing the number of components
+         * and instead reading them from the input file allows for more
```

Comment:
Do we need this constructor? I think the same one in AsciiDataLookup is only there for backwards compatibility.

Reply:
I don't know. Perhaps I can try with and without this constructor in the unit_test, and decide from there.